content/references/rippled-api/api-conventions/currency-formats.md

```diff
@@ -1,66 +1,112 @@
 
 ## Currency Codes
+[Currency Code]: #currency-codes
 
-All non-XRP currencies in the XRP Ledger have a 160-bit currency code. The [`rippled` APIs](rippled-api.html) map 3-character ASCII strings (case-sensitive) to 160-bit currency codes using a standard mapping. The currency code `XRP` is disallowed for issued currencies. Currencies with the same code can [ripple](rippling.html) across connected trustlines. Currency codes have no other behavior built into the XRP Ledger.
+{% include '_snippets/data_types/currency_code.md' %}
+<!--{#_ #}-->
 
-### Standard Currency Codes
 
-The standard currency mapping allocates the bits as follows:
+### Standard Currency Codes
 
-![Standard Currency Code Format](img/currency-code-format.png)
+The standard format for currency codes is a three-character string such as `USD`. This is intended for use with [ISO 4217 Currency Codes](http://www.xe.com/iso4217.php). The following rules apply:
 
-1. The first 8 bits must be `0x00`.
-2. The next 88 bits are reserved, and should be all `0`'s.
-3. The next 24 bits represent 3 characters of ASCII.
-    Ripple recommends using [ISO 4217](http://www.xe.com/iso4217.php) codes, or popular pseudo-ISO 4217 codes such as "BTC". However, any combination of the following characters is permitted: all uppercase and lowercase letters, digits, as well as the symbols `?`, `!`, `@`, `#`, `$`, `%`, `^`, `&`, `*`, `<`, `>`, `(`, `)`, `{`, `}`, `[`, `]`, and <code>&#124;</code>. The currency code `XRP` (all-uppercase) is reserved for XRP and cannot be used by issued currencies.
-4. The next 40 bits are reserved and should be all `0`'s.
+- Currency codes must be exactly 3 ASCII characters in length. The following characters are permitted: all uppercase and lowercase letters, digits, as well as the symbols `?`, `!`, `@`, `#`, `$`, `%`, `^`, `&`, `*`, `<`, `>`, `(`, `)`, `{`, `}`, `[`, `]`, and <code>&#124;</code>.
+- Currency codes are case-sensitive.
+- The currency code `XRP` (all-uppercase) is disallowed. Real XRP typically does not use a currency code in the XRP Ledger protocol.
 
-Usually, XRP amounts are not specified with currency codes. In the rare case that a field specifies a currency code for XRP, the currency code's binary format is all zeroes.
+At the protocol level, this format is [serialized](serialization.html#currency-codes) into a 160-bit binary value starting with `0x00`.
 
 ### Nonstandard Currency Codes
 
-You can also issue currency of other types by using a 160-bit (40-character) hexadecimal string such as `015841551A748AD2C1F76FF6ECB0CCCD00000000` as the currency code. To prevent this from being treated as a different currency code type, the first 8 bits MUST NOT be `0x00`.
+You can also issue currency of other types by using a 160-bit (40-character) hexadecimal string such as `015841551A748AD2C1F76FF6ECB0CCCD00000000` as the currency code. To prevent this from being treated as a "standard" currency code, the first 8 bits MUST NOT be `0x00`.
```

Comment:
¯\_(ツ)_/¯

You can make up your own 3-character code if you want. If you're worried about it getting confusing later because it clashes with an actual ISO code, some of the characters available for the XRPL's 3-character codes aren't allowed by the _actual_ ISO spec, so you can pick one that'll never be a valid ISO code. Or you could use the 160-bit option if you want to encode more info (as stated above).

Basically the only downsides to picking the same code that someone else uses for a completely different asset is that if a trader wants to hold both assets in the XRPL's decentralized exchange, they have to be careful not to accidentally enable [rippling](https://xrpl.org/rippling.html) between the two. (Through rippling, third parties can potentially cause the trader to passively trade the two same-name assets at a 1:1 rate.) But rippling is off by default so that shouldn't generally happen.